cmd/operator-sdk/bundle/validate.go

```diff
@@ -132,33 +169,75 @@ func (c bundleValidateCmd) run() (err error) {
 		fields["container-tool"] = c.imageBuilder
 	}
 	logger := log.WithFields(fields)
-	if viper.GetBool(flags.VerboseOpt) {
-		log.SetLevel(log.DebugLevel)
+	if c.isOutputJSON() {
+		// Log as JSON instead of the default ASCII formatter.
+		log.SetFormatter(&log.JSONFormatter{})
+	} else {
+		if viper.GetBool(flags.VerboseOpt) {
+			log.SetLevel(log.DebugLevel)
+		}
 	}
-
 	val := bundle.NewImageValidator(c.imageBuilder, logger)
-
 	// Pull image if a tag was passed.
 	if c.imageTag != "" {
-		logger.Info("Unpacked image layers")
+		msg := "Unpacked image layers"
+		if c.isOutputJSON() {
+			jsonOutput.addLog(msg, nil)
+		} else {
+			logger.Info(msg)
+		}
 		err = val.PullBundleImage(c.imageTag, dir)
 		if err != nil {
-			logger.Fatalf("Error to unpacking image: %v", err)
+			msg := "error to unpacking image"
+			if c.isOutputJSON() {
+				jsonOutput.printFatalJSONWith(msg, err)
+			}
+			logger.Fatalf(msg+": %v", err)
 		}
 	}
 
 	// Validate bundle format.
 	if err = val.ValidateBundleFormat(dir); err != nil {
-		logger.Fatalf("Bundle format validation failed: %v", err)
+		msg := "Bundle format validation failed"
+		if c.isOutputJSON() {
+			if valErr, isValErr := err.(bundle.ValidationError); isValErr {
+				for _, err := range valErr.Errors {
+					jsonOutput.addLog(msg, err)
+				}
+				jsonOutput.printFatalJSON()
+			}
+			jsonOutput.printFatalJSONWith(msg, err)
+		}
+		logger.Fatalf(msg+": %v", err)
 	}
 
 	// Validate bundle content.
 	manifestsDir := filepath.Join(dir, bundle.ManifestsDir)
 	if err = val.ValidateBundleContent(manifestsDir); err != nil {
-		logger.Fatalf("Bundle content validation failed: %v", err)
+		msg := "Bundle content validation failed"
+		if c.isOutputJSON() {
+			if valErr, isValErr := err.(bundle.ValidationError); isValErr {
+				for _, err := range valErr.Errors {
+					jsonOutput.addLog(msg, err)
+				}
+				jsonOutput.printFatalJSON()
+			}
+			jsonOutput.printFatalJSONWith(msg, err)
+		}
+		logger.Fatalf(msg+": %v", err)
 	}
 
-	logger.Info("All validation tests have completed successfully")
-
+	msg := "All validation tests have completed successfully"
+	if c.isOutputJSON() {
+		jsonOutput.addLog(msg, nil)
+		jsonOutput.printJSON()
+	} else {
+		logger.Info(msg)
+	}
```

Comment:
This and some other `if ... else ...` is making the code a little bit dirty. Maybe we need a wrapper over the `logrus` package used?

Reply:
I do not like the need to use the if .. else either .. however .. I am not sure if I catcher your suggestion for this specific scenario.